sage-ios/SAGE/SAGE/AnnouncementsDetailView.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  AnnouncementsDetailView.swift
+//  SAGE
+//
+//  Created by Erica Yin on 11/7/15.
+//  Copyright © 2015 Cal Blueprint. All rights reserved.
+//
+
+import Foundation
+
+class AnnouncementsDetailView: UIView {
+    var scrollView = UIScrollView()
+    let announcementUserImg = UIImageView()
+    var announcementUserName = UILabel()
+    var announcementTitle = UILabel()
+    var announcementTime = UILabel()
+    var announcementMessage = UILabel()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        scrollView.alwaysBounceVertical = true
+        self.addSubview(announcementUserImg)
+        self.addSubview(announcementUserName)
+        self.addSubview(announcementTime)
+        self.addSubview(announcementTitle)
+        self.addSubview(announcementMessage)
+        setUpViewStyle()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    func formatDate(date:NSDate) -> String {
+        let dateFormatter = NSDateFormatter()
+        dateFormatter.dateStyle = NSDateFormatterStyle.LongStyle
+        return dateFormatter.stringFromDate(date)
+    }
+    
+    func setupWithAnnouncement(announcement: Announcement) {
+        let image = UIImage(named: "UserImage.jpg")
+        self.announcementUserImg.image = image
+        self.announcementTitle.text = announcement.title
+        self.announcementMessage.text = announcement.text
+        self.announcementTime.text = self.formatDate((announcement.timeCreated)!)
+        var announcementTo = "Everyone"
+        if (announcement.school != nil) {
+            announcementTo = (announcement.school?.name)!
+        }
+        let boldString = [NSFontAttributeName: UIFont.getSemiboldFont(14.0)]
+        let nameLength = (announcement.sender?.firstName)!.characters.count + (announcement.sender?.lastName)!.characters.count + 1
+        let toIndex = nameLength + 4
+        let toLength = announcementTo.characters.count
+        let announcementString = NSMutableAttributedString(string: (announcement.sender?.firstName)! + " " + (announcement.sender?.lastName)! + " to " + announcementTo)
+        announcementString.addAttributes(boldString, range: NSRange(location: 0, length: nameLength))
+        announcementString.addAttributes(boldString, range: NSRange(location: toIndex, length: toLength))
+        self.announcementUserName.attributedText = announcementString
+        layoutSubviews()
+    }
+    
+    
+    func setUpViewStyle() {
+        announcementUserImg.setHeight(32)
```

Comment:
actually can you make this 32pt a constant in UIConstants? this will be a shared dimension for most images in tablview cells. make sure to the corner radius for the images are also this constant /2